pandas/io/common.py

```diff
@@ -82,6 +87,8 @@ def _expand_user(filepath_or_buffer):
     """
     if isinstance(filepath_or_buffer, str):
         return os.path.expanduser(filepath_or_buffer)
+    elif isinstance(filepath_or_buffer, pathlib.Path):
```

Comment:
We weren't expanding Path objects with existing code so this should handle that as well. Reviewing where to add test for this

Reply:
Might split this off into a separate issue/PR if I can find a bug using Path objects not expanding